### PR TITLE
readme: clarify host lifecycle behaviour

### PR DESCRIPTION
**Pull request by jhart0**
Doc only update clarifying lifecycle behavior for host containers

By submitting this pull request, I agree that this contribution is dual-licensed under the terms of both the Apache License, version 2.0, and the MIT license.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1306,6 +1306,7 @@ There are a few important caveats to understand about host containers:
 * They run in a separate instance of containerd than the one used for orchestrated containers like Kubernetes pods.
 * They're not updated automatically. You need to update the `source` and commit those changes.
 * If you set `superpowered` to true, they'll essentially have root access to the host.
+* If the container exits (whether successfully or not), it will always restart after 45 seconds
```
Comment (stmcginnis) [inline README.md]: Thanks, this is good clarification. A minor suggestion on the wording, but looks good!

```suggestion
* If the container exits for any reason, Bottlerocket will attempt to restart it after 45 seconds.
```

**Review by stmcginnis — APPROVED** [review]
Thanks @jhart0! 🎉

**Comment by jpculp** [comment]
Thanks for this! Before we merge, would you be willing to squash the commits under something like "readme: clarify host lifecycle behaviour"? If you would prefer we can also do it on our end.

**Comment by jhart0** [comment]
> would you be willing to squash the commits under something like "readme: clarify host lifecycle behaviour"

done, thanks!

**Review by jpculp — APPROVED** [review]

**Pull request merged**